rts/Lua/LuaUnitDefs.cpp

```diff
@@ -357,14 +357,15 @@ static int WeaponsTable(lua_State* L, const void* data)
 {
 	const auto& udWeapons = *reinterpret_cast<const decltype(UnitDef::weapons)*>(data);
 
-	lua_newtable(L);
+	lua_createtable(L, /*narr=*/udWeapons.size() + LUA_WEAPON_BASE_INDEX,
```

Comment:
is it really +INDEX? this is just an offset for indice values but there should still be just size() of them, starting from 1